examples/run-examples.sh

```diff
@@ -1,36 +1,148 @@
 #!/bin/bash
 # Testscript running all examples from example-list.txt
 #
-# Copyright (c) 2017      Christoph Niethammer <niethammer@hlrs.de>
+# Copyright (c) 2017-2024 Christoph Niethammer <niethammer@hlrs.de>
 #
 
-MARDYN_EXE="mpirun -np 1 $PWD/../src/MarDyn"
-MARDYN_OPTIONS="--steps 10 -v"
+# Some default values
+MARDYN_EXE="$PWD/../src/MarDyn"
+MARDYN_ARGS="--steps 20 -v"
+MPIRUN_EXE="mpirun"
+MPIRUN_ARGS="-n 4 --oversubscribe"
 EXAMPLE_LIST_FILE=${EXAMPLE_LIST_FILE:=example-list.txt}
 LOGFILE=${LOGFILE:=$PWD/run-examples.log}
 
-#mapfile -t all_examples < $EXAMPLE_LIST_FILE # requires bash4 ...
-declare -a all_examples
-while IFS= read -r; do all_examples+=("$REPLY"); done < $EXAMPLE_LIST_FILE
+TEMP=$(getopt -o vh --long "help,inputlist:,logfile:,mardyn_args:,mardyn_exe:,mpirun_args:,mpirun_exe:,verbose" -- $@)
 
+if [ $? -ne 0 ]; then
+  echo "Error parsing commandline"
+  exit 1
+fi
+
+eval set -- "$TEMP"
+unset TEMP
+
+function print_help() {
+cat <<EOF
+$(basename $0) is a script to execute MarDyn examples from a inputfile list file
+
+A inputlist file contains one path per line to a configuration file to be passed to MarDyn.
+MarDyn is executed with one configuration file at a time.
+In the inputlist file empty lines and lines starting with '#' are ignored.
+
+Usage:
+   $0 [OPTIONS ...]
+   $0 [-h|--help]
+
+Options:
+ -i,--inputlist   <FILE>    path to an inputfile list
+    --logfile     <FILE>    path for the logfile
+    --mpirun_exe  <mpirun>  mpirun command or path to mpirun executable
+    --mpirun_args <ARGS>    arguments to be passed to the mpirun command
+    --mardyn_exe  <MarDyn>  path to a MarDyn executable
+    --mardyn_args <ARGS>    arguments to be passed to MarDyn for each input file
+ -v,--verbose               verbose output
+ -h,--help                  show this help
+
+EOF
+}
+
+while true; do
+  case "$1" in
+    '--logfile')
+      LOGFILE="$2"
+      shift 2
+      continue
+      ;;
+    '--mardyn_exe')
+       MARDYN_EXE="$2"
+       shift 2
+       continue
+       ;;
+    '--mardyn_args')
+       MARDYN_EXE="$2"
+       shift 2
+       continue
+       ;;
+    '--mpirun_exe')
+       MARDYN_EXE="$2"
+       shift 2
+       continue
+       ;;
+    '--mpirun_args')
+       MARDYN_EXE="$2"
+       shift 2
+       continue
```

Comment:
Why does it store all the arguments in `MARDYN_EXE` ? This is imho wrong. Please test all options of this script before merging to master!
```suggestion
    '--mardyn_args')
       MARDYN_ARGS="$2"
       shift 2
       continue
       ;;
    '--mpirun_exe')
       MPIRUN_EXE="$2"
       shift 2
       continue
       ;;
    '--mpirun_args')
       MPIRUN_ARGS="$2"
       shift 2
       continue
```

Reply:
Dam ... will fix!